contracts/receivables/BaseCreditPoolReceivable.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import {BaseStructs as BS} from "../libraries/BaseStructs.sol";
+import "../BaseCreditPool.sol";
+
+/**
+ * @title BaseCreditPoolReceivable
+ * @dev Contract that represents an ERC721 receivable payable to Huma's BaseCreditPool
+ */
+contract BaseCreditPoolReceivable is ERC721Enumerable, ERC721URIStorage, AccessControl {
```

Comment:
off-chain receivables is evolving quickly in the industry. I am sure we will discover lots of new things that we do not know today. It will be prudent to make the contract upgradable.

Reply:
Will do